app/src/main/java/com/keylesspalace/tusky/ViewVideoActivity.kt

```diff
@@ -0,0 +1,191 @@
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.support.v4.content.FileProvider
+import android.support.v7.widget.Toolbar
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import android.view.MotionEvent
+import android.view.View
+import android.webkit.MimeTypeMap
+import android.widget.MediaController
+import android.widget.ProgressBar
+import android.widget.VideoView
+
+import com.keylesspalace.tusky.util.MediaUtils
+
+import java.io.File
+
+import com.keylesspalace.tusky.BuildConfig.APPLICATION_ID
+
+class ViewVideoActivity: BaseActivity() {
+
+    private val handler = Handler(Looper.getMainLooper())
+    lateinit var toolbar: Toolbar
+    private lateinit var url: String
+    private lateinit var statusID: String
+    private lateinit var statusURL: String
+
+    companion object {
+        private const val TAG = "ViewVideoActivity"
+        const val URL_EXTRA = "url"
+        const val STATUS_ID_EXTRA = "statusID"
+        const val STATUS_URL_EXTRA = "statusURL"
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_view_video)
+
+        val progressBar = findViewById<ProgressBar>(R.id.video_progress)
+        val videoView = findViewById<VideoView>(R.id.video_player)
+
+        toolbar = findViewById(R.id.toolbar)
+        setSupportActionBar(toolbar)
+        val bar = supportActionBar
+        if (bar != null) {
+            bar.title = null
+            bar.setDisplayHomeAsUpEnabled(true)
+            bar.setDisplayShowHomeEnabled(true)
+        }
+        toolbar.setOnMenuItemClickListener {item ->
+            val id = item.itemId
+            when (id) {
+                R.id.action_download -> downloadFile(url)
+                R.id.action_open_status -> onOpenStatus()
+                R.id.action_share_media -> shareVideo()
+            }
+            true
+        }
+
+        url = intent.getStringExtra(URL_EXTRA)
+        statusID = intent.getStringExtra(STATUS_ID_EXTRA)
+        statusURL = intent.getStringExtra(STATUS_URL_EXTRA)
+
+        videoView.setVideoPath(url)
+        val controller = MediaController(this)
+        controller.setMediaPlayer(videoView)
+        videoView.setMediaController(controller)
+        videoView.requestFocus()
+        videoView.setOnPreparedListener { mp ->
+            progressBar.visibility = View.GONE
+            mp.isLooping = true
+            hideToolbarAfterDelay()
+        }
+        videoView.start()
+
+        videoView.setOnTouchListener { _, event ->
+            if (event.action == MotionEvent.ACTION_DOWN) {
+                handler.removeCallbacksAndMessages(null)
+                toolbar.animate().cancel()
+                toolbar.alpha = 1.0f
+                toolbar.visibility = View.VISIBLE
```

Comment:
you can use the extension functions I created for this `toolbar.show()`

Reply:
To be clear, you meant for setting the actual visibility property, right? Not to remove the animations entirely?